source/patterns/@aws-solutions-constructs/resources/kms-key-policy-updater/tests/kms-key-policy-updater.test.ts

```diff
@@ -0,0 +1,156 @@
+/**
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+
+import { mockClient } from "aws-sdk-client-mock";
+import { KMSClient, DescribeKeyCommand, KeyManagerType, GetKeyPolicyCommand, PutKeyPolicyCommand } from "@aws-sdk/client-kms";
+import { handler } from "../lib";
+
+const kmsMock = mockClient(KMSClient);
+
+beforeEach(() => {
+  kmsMock.reset();
+});
+
```

Comment:
George started the conversation with a test here that called the function twice and then checked that there was not a duplicate policy added.

Reply:
Ack, addressed this in d50315298ecb4a072a128656ad57a38761b9aab6 and added tests to simulate the custom resource short-circuiting if a matching sid is found in the array of key policy statements.